### ISSUE TITLE
Asynchronous interface for sqlite

### ISSUE THREAD

**Issue by vxgmichel** [Scille/parsec-cloud#635]
While reading or writing to the persistent storage, a good amount of time is spent performing sqlite requests. This can take up to hundreds of milliseconds for large blocks of data (typically 512KB), especially when several requests are performed without yielding to the trio loop in between.

It would be good let trio run while performing those requests in an other thread. There might be libraries to help with that.

For information, typical warnings that one might get when importing a large file into a workspace:
```
2019-08-23 08:50:40 [warning  ] Task step took too long        [parsec.monitoring] duration=0.27977991104125977 task_name=parsec.core.sync_monitor.monitor_sync
  File "/home/max/work/parsec-cloud/parsec/core/sync_monitor.py", line 393, in monitor_sync
    await _monitor_sync_online(user_fs, event_bus)
  File "/home/max/work/parsec-cloud/parsec/core/sync_monitor.py", line 369, in _monitor_sync_online
    wait_times.append(await _ctx_tick(ctx))
  File "/home/max/work/parsec-cloud/parsec/core/sync_monitor.py", line 329, in _ctx_tick
    return await ctx.tick()
  File "/home/max/work/parsec-cloud/parsec/core/sync_monitor.py", line 205, in tick
    await self._sync(entry_id)
  File "/home/max/work/parsec-cloud/parsec/core/sync_monitor.py", line 243, in _sync
    await self.workspace.sync_by_id(entry_id, recursive=False)
  File "/home/max/work/parsec-cloud/parsec/core/fs/workspacefs/workspacefs.py", line 581, in sync_by_id
    manifest = await self._sync_by_id(entry_id, remote_changed=remote_changed)
  File "/home/max/work/parsec-cloud/parsec/core/fs/workspacefs/workspacefs.py", line 541, in _sync_by_id
    await self._upload_blocks(new_remote_manifest)
  File "/home/max/work/parsec-cloud/parsec/core/fs/workspacefs/workspacefs.py", line 454, in _upload_blocks
    await self.remote_loader.upload_block(access, data)
  File "/home/max/work/parsec-cloud/parsec/core/fs/remote_loader.py", line 236, in upload_block
    await self.backend_cmds.block_create(access.id, self.workspace_id, ciphered)
  File "/home/max/work/parsec-cloud/parsec/core/backend_connection/porcelain.py", line 40, in wrapper
    return await cmd(transport, *args, **kwargs)
  File "/home/max/work/parsec-cloud/parsec/core/backend_connection/cmds.py", line 351, in block_create
    block=block,
  File "/home/max/work/parsec-cloud/parsec/core/backend_connection/cmds.py", line 86, in _send_cmd
    await transport.send(raw_req)
  File "/home/max/work/parsec-cloud/parsec/api/transport.py", line 131, in send
    await self._net_send()
  File "/home/max/work/parsec-cloud/parsec/api/transport.py", line 75, in _net_send
    await self.stream.send_all(out_data)

2019-08-23 08:50:40 [warning  ] Task step took too long        [parsec.monitoring] duration=0.3900790214538574 task_name=parsec.core.gui.trio_thread.QtToTrioJob._run_fn
  File "/home/max/work/parsec-cloud/parsec/core/gui/trio_thread.py", line 66, in _run_fn
    self.ret = await self._fn(*self._args, **self._kwargs)
  File "/home/max/work/parsec-cloud/parsec/core/gui/files_widget.py", line 86, in _do_import
    await workspace_fs.write_bytes(dst, chunk, read_size)
  File "/home/max/work/parsec-cloud/parsec/core/fs/workspacefs/workspacefs.py", line 355, in write_bytes
    _, fd = await self.transactions.file_open(path, "w")
  File "/home/max/work/parsec-cloud/parsec/core/fs/workspacefs/entry_transactions.py", line 429, in file_open
    async with self._lock_manifest_from_path(path) as manifest:
  File "/home/max/work/parsec-cloud/venv/lib/python3.6/site-packages/async_generator/_util.py", line 34, in __aenter__
    return await self._agen.asend(None)
```

**Comment by touilleMan** [comment]
@vxgmichel  implemented in #675 ?

**Comment by vxgmichel** [comment]
> The storage interface is not connected to anything asynchronous for the moment (the synchronous sqlite3 library is still in use) but it will be easier to implement it in a later PR.

**Comment by vxgmichel** [comment]
Fixed with #743
